src/lib/components/PrismicImage.svelte

```diff
@@ -1,34 +1,63 @@
 <script lang="ts">
-	import * as prismicT from "@prismicio/types";
+	import * as prismicH from "@prismicio/helpers"
+	import type * as prismicT from "@prismicio/types";
+	
+	/**
+	 * A Prismic Image field.
+	 */
 	export let field: prismicT.ImageField;
-	let className: string;
 
-	// creates a `class` property, even
-	// though it is a reserved word
-	export { className as class };
+	/**
+	 * An object of Imgix URL API parameters to transform the image.
+	 *
+	 * See: https://docs.imgix.com/apis/rendering
+	 */
+	export let imgixParams: Parameters<typeof prismicH.asImageSrc>[1] | null | undefined = null;
 
-	const usePrismicImage = (
-		field: prismicT.ImageField,
-	): {
-		url: string;
-		width?: number;
-		height?: number;
-		alt?: string;
-		copyright: string | undefined;
-	} => {
-		field.url;
-		return {
-			url: field.url || "",
-			alt: field.alt || "",
-			width: field.dimensions?.width || undefined,
-			height: field.dimensions?.height || undefined,
-			copyright: field.copyright || undefined,
-		};
-	};
+	/**
+	 * Widths used to build a `srcset` value for the Image field.
+	 *
+	 * If a `widths` prop is not given or `"defaults"` is passed, the
+	 * following widths will be used: 640, 750, 828, 1080, 1200, 1920, 2048,
+	 * 3840.
+	 *
+	 * If the Image field contains responsive views, each responsive view
+	 * can be used as a width in the resulting `srcset` by passing
+	 * `"thumbnails"` as the `widths` prop.
+	 */
+	export let widths: NonNullable<Parameters<typeof prismicH.asImageWidthSrcSet>[1]>["widths"] | null | undefined = null;
+				
+	
+	/**
+	 * HTML class attribute. Use a `:global()` selector to target the class.
+	 */
+	let className: string | null | undefined = null;
+	export { className as class };
 
-	const image = usePrismicImage(field);
+	const params = {}
+	if(widths) Object.assign(params, { widths })
+	if(imgixParams) Object.assign(params, imgixParams)
```

Comment:
This block of code can be simplified to:

```ts
const params = { widths, ...imgixParams }
```

To make this work, `null` needs to be removed as possible types for the `widths` and `imgixParams` props.

Since there isn't significant work being done in this refactored code, it can be inlined where it's used:

```ts
{@const { src, srcset } = prismicH.asImageWidthSrcSet(field, { widths, ...imgixParams })}
```

As a stylistic change, what do you think about doing all computation in `<script>`? That would require moving `prismicH.asImageWidthSrcSet` outside the template and into `<script>`.

Reply:
I simplified and inlined this, but then elaborated it to support pixel densities. I think it still works, but let me know if you think it is now too verbose to be inside the `<script>` tag.

Reply:
https://github.com/prismicio/prismic-svelte/blob/97ee8c6f2352195b8cc80f4a1250b6c3306156c2/src/lib/components/PrismicImage.svelte#L118

This line can be simplified to:

```ts
pixelDensities === "defaults" ? undefined : pixelDensities,
```

Similarly, the `asImageWidthSrcSet()` call can be simplified to:

```ts
widths: widths === "defaults" ? undefined : widths,
```

We can then remove the `defaultWidths` constant.

We can make this simplification because `asPixelDensitySrcSet()` and `asImageWidthSrcSet()` fall back to their internal default values if `pixelDensities` or `widths` is not defined. This lets us change the default values in `@prismicio/helpers` and trickle down those changes to `@prismicio/svelte`, `@prismicio/react`, etc. automatically.

Reply:
✅